src/OpenTelemetry/Metrics/MeterProviderBuilderBase.cs

```diff
@@ -88,19 +89,51 @@ internal MeterProviderBuilder AddReader(MetricReader reader)
 
         internal MeterProviderBuilder AddView(string instrumentName, string name)
         {
-            // TODO: Actually implement view.
-            return this;
+            var metricConfig = new MetricStreamConfiguration() { Name = name };
+            return this.AddView(instrumentName, metricConfig);
         }
 
-        internal MeterProviderBuilder AddView(string instrumentName, MetricStreamConfiguration aggregationConfig)
+        internal MeterProviderBuilder AddView(string instrumentName, MetricStreamConfiguration metricStreamConfiguration)
         {
-            // TODO: Actually implement view.
-            return this;
+            Func<Instrument, MetricStreamConfiguration> viewConfig = (instrument) =>
+            {
+                var needRegex = false;
+                if (instrumentName.IndexOf('*') != -1)
+                {
+                    needRegex = true;
+                }
+
+                if (needRegex)
+                {
+                    var pattern = '^' + Regex.Escape(instrumentName).Replace("\\*", ".*");
```

Comment:
This will create string for every single instrument, which is very costly.
Need to extract the logic we don't repeat things.